ompi/mca/fbtl/posix/fbtl_posix_preadv.c

```diff
@@ -160,11 +160,16 @@ ssize_t mca_fbtl_posix_preadv_datasieving (ompio_file_t *fh, struct flock *lock,
         }
         
         size_t sstart = (size_t)fh->f_io_array[startindex].offset;
-        size_t slen=0;
+        size_t slen=0, maxlen=0;
+        int maxindex = startindex;
 
         for ( j = startindex; j < fh->f_num_of_io_entries; j++ ) {
             endindex = j;
             slen = ((size_t)fh->f_io_array[j].offset + fh->f_io_array[j].length) - sstart;            
+            if (slen > maxlen) {
```

Comment:
I'm not sure this is correct: the next branch checks `slen` and essentially discards this iteration. Shouldn't the order of the two branches be changed?

Reply:
I think it should be correct, since if slen < maxlen, the second if statement would have been either true in the previous iteration, or will also not be true in this iteration.

Reply:
actually, you might be right. Thanks!